src/analyze.ts

```diff
@@ -399,17 +409,17 @@ module.exports = async function (id, code, job) {
   handleWrappers(ast);
   ({ ast = ast, scope = scope } = handleSpecialCases({ id, ast, scope, emitAsset: path => assets.add(path), emitAssetDirectory, job }) || {});
 
-  function backtrack (self, parent) {
+  function backtrack (self: WalkerContext, parent: Node) {
```

Comment:
I found that `parent` can be undefined according to the typescript declarations in `walk`.

```js
  walk(ast, {
    enter (node, parent) {
     // ...
     }
```

Did you know this? How should we handle that case for `backtrack()` and `isIdentifierRead()`?

Reply:
`parent` can only be undefined when `node` is a `Program`, and we never call `backtrack` on a `Program` node so this can't happen. Feel free to just add a forced typing as necessary here.

Reply:
Just to be clear, the only time `parent` is undefined is when `node.type === 'Program'`, correct?

Reply:
Yes exactly.